### ISSUE TITLE
JSON parser decodes escapes in selectors

### ISSUE THREAD

**Issue by glyn** [jsonpath-standard/jsonpath-compliance-test-suite#34]
@akshaymankar [raised](https://social.coop/@akshay/110235938714895279) the following issue:

Hi, I finally found time to work on [making [jsonpath-hs](https://github.com/akshaymankar/jsonpath-hs) pass the CTS]. I think the cts.json needs to double escape all the tests [with] selectors which encode characters using the `\u` notation. Otherwise the JSON parser which parses the test already decodes them giving no chance to the jsonpath parser.

**Comment by gregsdennis** [comment]
Are we talking about updating

```json
{
  "name": "name selector, double quotes, embedded U+0000",
  "selector": "$[\"\u0000\"]",
  "invalid_selector": true
}
```

to

```json
{
  "name": "name selector, double quotes, embedded U+0000",
  "selector": "$[\"\\u0000\"]",
  "invalid_selector": true
}
```

That means the un-JSON'd path is `$["\u0000"]` instead of `$[""]` (which isn't rendered here but should have a `0`-char in the string).  Do we want to test both, maybe?

**Comment by glyn** [comment]
Yes, I think we need to test both those situations. If we don't already, we should also test the use of a Unicode escape in a query to represent a character that is not escaped in the JSON.
